embassy-macros/src/macros/task.rs

```diff
@@ -45,8 +79,10 @@ pub fn run(args: syn::AttributeArgs, f: syn::ItemFn) -> Result<TokenStream, Toke
         },
     }
 
-    if pool_size < 1 {
-        ctxt.error_spanned_by(&f.sig, "pool_size must be 1 or greater");
+    if let PoolSize::Literal(pool_size) = pool_size {
+        if pool_size < 1 {
```

Comment:
I wonder if we could/should get rid of this check?
- If you're controlling the pool size based on some global const based on some calculation or Cargo feature, it might make sense to set it as zero in some build configs. (like, use 0 TCP server tasks if this product variant has no Ethernet)
- It's weird that zero length is allowed via a const, but not if you put the literal 0 in the meta.

This would allow parsing the attr as just "expression", instead of "literal or expression", which would simplify this a lot I think?

Reply:
> It's weird that zero length is allowed via a const

Yeah, that is actually an unintended side effect. I don't think proc macros can access the value of a constant. This also means we can't reject paths to non-constants here.

I guess I can try arbitrary expressions. Not sure how much simpler things can get, we'll see.

Reply:
ah I missed you're parsing it as "literal or path", not "literal or expression" Yeah, I'm not sure if arbitrary expressions is a good idea. Even if it still accepts only literals or paths, removing the check should allow removing the enum which simplifies things a bit at least :sweat_smile: